File: RaptureXMLTests/DeepTests.m
//
//  DeepTests.m
//  RaptureXML
//
//  Created by John Blanco on 9/24/11.
//  Copyright (c) 2011 Rapture In Venice. All rights reserved.
//

#import <UIKit/UIKit.h>
#import <SenTestingKit/SenTestingKit.h>
#import "RXMLElement.h"

@interface DeepTests : SenTestCase {
}

@end



@implementation DeepTests

- (void)testQuery {
    RXMLElement *rxml = [RXMLElement elementFromXMLFile:@"players.xml"];
    __block NSInteger i;
    
    // count the players
    i = 0;
    
    [rxml iterate:@"players.player" with: ^(RXMLElement *e) {
        i++;
    }];    
    
    STAssertEquals(i, 9, nil);

    // count the first players' names
    i = 0;
    
    [rxml iterate:@"players.player.name" with: ^(RXMLElement *e) {
        i++;
    }];    
    
    STAssertEquals(i, 1, nil);

    // count the coaches
    i = 0;
    
    [rxml iterate:@"players.coach" with: ^(RXMLElement *e) {
        i++;
    }];    
    
    STAssertEquals(i, 1, nil);
}

@end

File: Swift/RaptureXMLTests/DeepTests.swift
//
//  DeepTests.swift
//  RaptureXML-Swift
//
//  Created by Brett Lamy on 2/12/15.
//  Copyright (c) 2015 Brett Lamy. All rights reserved.
//

import UIKit
import XCTest

class DeepTests: XCTestCase {

    func testQuery() {
        let rxml = RXMLElement(fromXMLFile:"players.xml");
        
        // count the players
        var i = 0;
        
        rxml.iterate("players.player", usingBlock: { (e : RXMLElement!) -> Void in
            i += 1;
        });
        
        XCTAssertEqual(i, 9);
        
        // count the first player's name
        i = 0;
        
        rxml.iterate("players.player.name", usingBlock: { (e : RXMLElement!) -> Void in
            i += 1;
        });
        
        XCTAssertEqual(i, 1);
        
        // count the coaches
        i = 0;
        
        rxml.iterate("players.coach", usingBlock: { (e : RXMLElement!) -> Void in
            i += 1;
        });
        
        XCTAssertEqual(i, 1);
    }
}
